client/state/themes/actions.js

```diff
@@ -106,13 +106,53 @@ export function receiveTheme( theme, siteId ) {
  * @return {Object}        Action object
  */
 export function receiveThemes( themes, siteId ) {
-	return {
-		type: THEMES_RECEIVE,
-		themes,
-		siteId
+	return ( dispatch, getState ) => {
+		const { filteredThemes } = filterThemes( getState, themes, siteId );
+		dispatch( {
+			type: THEMES_RECEIVE,
+			themes: filteredThemes,
+			siteId
+		} );
+	};
+}
```

Comment:
Aha, got you. I hadn't fully grasped this. I think your theory that wpcom themes are making it into the jp query when the filter occurs before the [site info has fully loaded](https://github.com/Automattic/wp-calypso/blob/master/client/state/themes/actions.js#L171) is the only credible explanation.

I'm going to close this PR and look at what we might be able to do about that.